includes/reports/class-sensei-no-users-table-relationship.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * File containing the Sensei_No_Users_Table_Relationship class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class that handles a special case of environments that have the users table in a separate
+ * database from the other tables.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_No_Users_Table_Relationship {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_No_Users_Table_Relationship constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
```

Comment:
I believe you can use type hints on PHP 7.2 already, so this code becomes this:

```suggestion
	public static function instance() : self {
```